Cannot implicitly convert type `byte' to `bool' when binding dear imgui

##### Brief Description

When I try to bind [dear imgui](https://github.com/ocornut/imgui/blob/master/imgui.h), the generated C# code is not compilable. Problem is caused by https://github.com/ocornut/imgui/blob/28a31997b76c59dba9d6d2ba03c5c1cf7b3a6ce6/imgui.h#L855

OS: Windows 7

##### Used headers

https://github.com/ocornut/imgui/blob/28a31997b76c59dba9d6d2ba03c5c1cf7b3a6ce6/imgui.h

##### Used settings


Target: MSVC

Other settings 


##### Stack trace or incompilable generated code

```cs
        public bool[] MouseDown
        {
            get
            {
                bool[] __value = null;
                if (((global::Engine.ImGuiIO.__Internal*) __Instance)->MouseDown != null)
                {
                    __value = new bool[5];
                    for (int i = 0; i < 5; i++)
                        __value[i] = ((global::Engine.ImGuiIO.__Internal*) __Instance)->MouseDown[i];
                }
                return __value;
            }

            set
            {
                if (value != null)
                {
                    for (int i = 0; i < 5; i++)
                        ((global::Engine.ImGuiIO.__Internal*) __Instance)->MouseDown[i] = value[i];
                }
            }
        }
```

```error CS0029: Cannot implicitly convert type `byte' to `bool'```

--- a/src/Generator/Generators/CSharp/CSharpMarshal.cs
+++ b/src/Generator/Generators/CSharp/CSharpMarshal.cs
@@ -115,7 +115,10 @@ namespace CppSharp.Generators.CSharp
                         }
                         else
                         {
-                            if (arrayType.IsPrimitiveType(PrimitiveType.Char) &&
+                            if (arrayType.IsPrimitiveType(PrimitiveType.Bool))
+                                supportBefore.WriteLineIndent($@"{value}[i] = {
+                                    Context.ReturnVarName}[i] != 0;");
+                            else if (arrayType.IsPrimitiveType(PrimitiveType.Char) &&
                                 Context.Context.Options.MarshalCharAsManagedChar)
                                 supportBefore.WriteLineIndent($@"{value}[i] = global::System.Convert.ToChar({
                                     Context.ReturnVarName}[i]);");
@@ -514,7 +517,11 @@ namespace CppSharp.Generators.CSharp
                     }
                     else
                     {
-                        if (arrayType.IsPrimitiveType(PrimitiveType.Char) &&
+                        if (arrayType.IsPrimitiveType(PrimitiveType.Bool))
+                            supportBefore.WriteLineIndent($@"{
+                                Context.ReturnVarName}[i] = (byte)({
+                                Context.ArgName}[i] ? 1 : 0);");
+                        else if (arrayType.IsPrimitiveType(PrimitiveType.Char) &&
                             Context.Context.Options.MarshalCharAsManagedChar)
                             supportBefore.WriteLineIndent($@"{
                                 Context.ReturnVarName}[i] = global::System.Convert.ToSByte({

--- a/tests/CSharp/CSharp.Tests.cs
+++ b/tests/CSharp/CSharp.Tests.cs
@@ -285,8 +285,12 @@ public unsafe class CSharpTests : GeneratorTestFixture
     public void TestBooleanArray()
     {
         Foo foo = new Foo { A = 10 };
-        for (int i = 0; i < foo.Btest.Length; ++i)
-            foo.Btest[i] = i % 2 == 0;
+        var new_values = new bool[5];
+        for(int i = 0; i < new_values.Length; ++i)
+        {
+            new_values[i] = i % 2 == 0;
+        }
+        foo.Btest = new_values;
         Assert.AreEqual(true, foo.Btest[0]);
         Assert.AreEqual(false, foo.Btest[1]);
         Assert.AreEqual(true, foo.Btest[2]);

--- a/tests/CSharp/CSharp.Tests.cs
+++ b/tests/CSharp/CSharp.Tests.cs
@@ -282,6 +282,20 @@ public unsafe class CSharpTests : GeneratorTestFixture
     }
 
     [Test]
+    public void TestBooleanArray()
+    {
+        Foo foo = new Foo { A = 10 };
+        for (int i = 0; i < foo.Btest.Length; ++i)
+            foo.Btest[i] = i % 2 == 0;
+        Assert.AreEqual(true, foo.Btest[0]);
+        Assert.AreEqual(false, foo.Btest[1]);
+        Assert.AreEqual(true, foo.Btest[2]);
+        Assert.AreEqual(false, foo.Btest[3]);
+        Assert.AreEqual(true, foo.Btest[4]);
+    }
+
+
+    [Test]
     public void TestImplicitCtor()
     {
         Foo foo = new Foo { A = 10 };

--- a/tests/CSharp/CSharp.h
+++ b/tests/CSharp/CSharp.h
@@ -32,6 +32,8 @@ public:
     int operator ++();
     int operator --();
 
+    bool btest[5];
+
 protected:
     int P;
     TemplateInAnotherUnit<int> templateInAnotherUnit;

